cmd/limactl/copy.go

```diff
@@ -0,0 +1,75 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"os/user"
+	"strconv"
+	"strings"
+
+	"github.com/AkihiroSuda/lima/pkg/sshutil"
+	"github.com/AkihiroSuda/lima/pkg/store"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli/v2"
+)
+
+var copyCommand = &cli.Command{
+	Name:        "copy",
+	Aliases:     []string{"cp", "scp"},
+	Usage:       "Copy files between host and guest",
+	Description: "Prefix guest filenames with a colon. Example: limactl copy default :/etc/os-release .",
```

Comment:
Do we really need colon?

Reply:
We need some mechanism to tell which file is on the host, and which is on the guest. If we don't use the colon, how would you specify the direction:

```sh
# copy from guest to host
limactl cp default :/etc/hostname hostname
# copy from host to guest
limactl cp default file1 file2 file3 :.
```

Reply:
An alternative syntax: `limactl cp default:/etc/hostname hostname`.

This syntax can be also extended to `limactl cp inst1:/foo inst2:/foo` to copy files across instances.

This can be also extended to `limactl cp inst1 inst2` to copy instances, though it might be confusing.
But given that `limactl ls` lists instances and `limactl delete` deletes instances, `limactl cp inst1 inst2` might be probably acceptable.

Reply:
> This syntax can be also extended to `limactl cp inst1:/foo inst2:/foo` to copy files across instances.

I like it. I just did a manual test, and it seems to work:

```console
$ scp -i ~/.lima/_config/user -o NoHostAuthenticationForLocalhost=yes -3 scp://jan@127.0.0.1:60020//etc/hostname scp://jan@127.0.0.1:60022/myname
$ limactl shell default sh -c 'cat ~/myname'
lima-alpine
```

I was confused for a bit about the obscure syntax: the first slash after the port is just a separator and not part of the path. You need a double-slash to specify an absolute path. At least it means relative paths are still supported.

The copy still has to go through the host (using `-3`) and not directly between the instances because they use different ports.

I guess theoretically you could use `ssh` to run the `scp` command from within the source instances, but that seems like unnecessary complexity.

Reply:
I don't think it will be possible to use the `ControlPath` when we copy between instances, so this will complicate the logic some more...

I'll take a look into it tomorrow.

Reply:
Thanks, `limactl cp inst1:/foo inst2:/foo` can be implemented later, we just have to determine the CLI syntax here.

Reply:
I just implemented it:

```console
$ limactl cp alpine:/etc/os-release opensuse:release-info
$ limactl shell opensuse sh -c 'cat ~/release-info'
NAME="Alpine Linux"
ID=alpine
VERSION_ID=3.13.5
PRETTY_NAME="Alpine Linux v3.13"
HOME_URL="https://alpinelinux.org/"
BUG_REPORT_URL="https://bugs.alpinelinux.org/"
```

scp will not use the `ControlPath`, but I think that is fine; it is just an optimization.